phantazm-commons/src/main/java/com/github/phantazmnetwork/commons/AdventureConfigProcessors.java

```diff
@@ -0,0 +1,89 @@
+package com.github.phantazmnetwork.commons;
+
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.collection.ConfigNode;
+import com.github.steanky.ethylene.core.collection.LinkedConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import net.kyori.adventure.key.InvalidKeyException;
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.sound.Sound;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.jetbrains.annotations.NotNull;
+
+public final class AdventureConfigProcessors {
+    private AdventureConfigProcessors() {
+        throw new UnsupportedOperationException();
+    }
+
+    private static final ConfigProcessor<Key> key = new ConfigProcessor<>() {
+        @Override
+        public Key dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+            try {
+                //noinspection PatternValidation
+                return Key.key(ConfigProcessor.STRING.dataFromElement(element));
+            }
+            catch (InvalidKeyException keyException) {
+                throw new ConfigProcessException(keyException);
+            }
+        }
+
+        @Override
+        public @NotNull ConfigElement elementFromData(Key key) throws ConfigProcessException {
+            return ConfigProcessor.STRING.elementFromData(key.asString());
+        }
+    };
+
+    private static final ConfigProcessor<Component> component = new ConfigProcessor<>() {
+        @Override
+        public Component dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+            return MiniMessage.miniMessage().deserialize(ConfigProcessor.STRING.dataFromElement(element));
+        }
+
+        @Override
+        public @NotNull ConfigElement elementFromData(Component component) throws ConfigProcessException {
+            String string = MiniMessage.miniMessage().serialize(component);
+            return ConfigProcessor.STRING.elementFromData(string);
```

Comment:
general question about when to use ConfigProcessor.X vs manually serializing

Reply:
(actually, if you need to verify that a ConfigElement is indeed X, using ConfigProcessor.X.dataFromElement is probably a good idea)

Reply:
(not necessary in this example though)

Reply:
doesn't getXorThrow already do that

Reply:
not if you're expecting the ConfigElement itself to be a string